hamster_gtk/helpers.py

```diff
@@ -177,6 +180,23 @@ def decompose_raw_fact_string(text, raw=False):
     return result
 
 
+# [TODO]
+# Oncec LIB-251 has been fixed this should no longer be needed.
+def get_recent_activities(controller, start, end):
+    """Return a list of all activities logged in facts within the given timeframe."""
+    recent_activities = [fact.activity for fact in controller.facts.get_all(start=start, end=end)]
+    return OrderedSet(recent_activities)
```

Comment:
The activities are returned in a convoluted order. It should be reversed prior to removing duplicates.

Reply:
It should be `OrderedSet(reversed(recent_activities))`, otherwise you will get the activities in the order they were first used, not the last.

```
facts     1 2 1 3 2 1 3  output   reversed
current   x x   x       → 1 2 3  → 3 2 1
expected          x x x → 3 1 2  → 2 1 3
```

Reply:
Thank you for the clarification. To be honest, right now ``FactManager.get_all()`` does not commit to any ordering at all. I created an [issue with ``hamster-lib``](https://projecthamster.atlassian.net/browse/LIB-259) to improve this.
For now I will implement a workaround simple sorting call as part of the prelimery ``get_recent_activities`` helper which will hopefully address your concerns.